FSI/cylinderFlap/OpenFOAM-FEniCS/Solid/cyl-flap.py

```diff
@@ -88,7 +88,7 @@ def remaining_boundary(x, on_boundary):
 force_boundary = AutoSubDomain(remaining_boundary)
 
 # Initialize the coupling interface
-precice_dt = precice.initialize(coupling_boundary, mesh, V, dim, fixed_boundary=clamped_boundary_domain)
+precice_dt = precice.initialize(coupling_boundary, V, V, clamped_boundary_domain)
```

Comment:
Why do we need to pass "V" twice?

Reply:
The new proposed initialization of the adapter is: `initialize(self, coupling_subdomain, read_function_space=None, write_object=None, fixed_boundary=None)`. The function space `V` is passed twice because both read and write functions live on the same function space in this case. It is necessary to pass the function space twice, once as `read_function_space` and again as `write_object` so that the adapter knows it is a bi-directional coupling and the user wants to *read* and *write* data.
Initializing like: `precice.initialize(coupling_boundary, read_function_space=V, fixed_boundary=clamped_bounday_domain)` would mean the user wants to perform a read-only uni-directional coupling.

This new initialization basically provides a way to define uni-directional coupling. Does this make sense?

Reply:
OK, thanks for clarification. I think it's not perfect, but good enough for the moment. Let's maybe add a quick comment in the tutorials code.

Reply:
```suggestion
precice_dt = precice.initialize(coupling_boundary, read_function_space=V, write_object=V, fixed_boundary=clamped_boundary_domain)
```
I would suggest to use the keyword to make the actual usage of `V` more explicit. This is optional, but from my point of view improves readability of the tutorial.